ostree-upload-server.py

```diff
@@ -76,10 +85,27 @@ def get_queue(self):
     def join(self, timeout=None):
         return self.queue.join(timeout)
 
+class Counter:
+    def __init__(self):
+        self.count = 0
+
+    def __enter__(self):
+        self.count += 1
+        print("counter now " + str(self.count))
+        return self.count
+
+    def __exit__(self, type, value, traceback):
+        self.count -= 1
+        print("counter now " + str(self.count))
```

Comment:
I'm not sure how threading.Lock() works with gevent so I used gevent.lock.BoundedSemaphore(1)